File: pset2/vigenere.c
/*This program will encypt a message for you using Vigenere's cipher*/

#include <stdio.h>
#include <cs50.h>
#include <string.h> //header file for strlen function
#include <ctype.h>  //header file for isalpha, isupper, islower function

int main(int argc, string argv[])

{
    if (argc != 2)      //print error and exit if command-line does not contain exactly two strings
    {
        printf ("error\n");
        return 1;
    }
    
int u, l, keyl = strlen(argv[1]);       //variable declarations
string word, key = argv[1];

    for (int a = 0, j = 0; a < keyl; a++)       //for-loop is used to go through every letter of the key
    {
        if (!isalpha(argv[1][a]))       //checks if there are non-alphabetic chars, if so, error and exit
        {
            printf ("error\n");
            return 1;
        }
        else if (isupper(argv[1][a]))       //converts any uppercase letters to its respective alpha-index value (0-25)
        {
            key[j] = (argv[1][a]) - 'A';    //'A' = 97 in ascii index
            j++;                            //stores value in 'j' and proceeds to next char
        }
        else if (islower(argv[1][a]))       //converts any lowercase letters to its respective alpha-index value (0-25)
        {
            key[j] = (argv[1][a]) - 'a';    //'a' = 65 in ascii index
            j++;                            //stores value in 'j' and proceeds to next char
        }
    }
    
    printf ("plaintext: ");     //prompts user for message
    word = get_string();
    
    printf ("ciphertext: ");
    for (int i = 0, j = 0, length = strlen(word); i < length; i++)      //for-loop is used to go through every letter of the message
    {
        if (isalpha(word[i]) && isupper(word[i]))
        {
            u = (((word[i]-65)+key[j%keyl])%26);        //j%keyl: 'j' is the stored value calculated earlier, %keyl will Mod the length of the key and this will allow
            printf ("%c", u+65);                        //the program to keep looping over the key's values to cipher the message
            j++;
        }
        else if (isalpha(word[i]) && islower(word[i]))
        {
            l = (((word[i]-97)+key[j%keyl])%26);
            printf ("%c", l+97);
            j++;
        }
        else
        {
            printf ("%c", word[i]);     //if char is not a alphabetical letter, just print it
        }
    }
    printf ("\n");      //new line
    return 0;
}

File: pset6/pset6/vigenere.py
#Converting Vigenere in C to Python

import cs50
import sys

def main():
    
    key = []
    j = 0
    i = 0
    
    if len(sys.argv) != 2:
        print ("error1")
        exit(1)
    
    keyword = sys.argv[1]
    if str.isalpha(keyword) == False:
        print ("error2")
        exit(1)
        
    print ("plaintext: ", end="")
    msg = cs50.get_string()
    
    for c in keyword:
        if c.isupper():
            i = (ord(c) - 65)
            key.append(i)
        if c.islower():
            i = (ord(c) - 97)
            key.append(i)
            
    print ("ciphertext: ", end="")
    
    k = 0
    
    for d in msg:
        if d.isupper():
            j = (((ord(d) - 65) + (key[k%len(keyword)]))%26) + 65
            print (chr(j), end="")
            k += 1
        if d.islower():
            j = (((ord(d) - 97) + (key[k%len(keyword)]))%26) + 97
            print (chr(j), end="")
            k += 1
        else:
            print (d, end="")
            
    print ("\n")
            
if __name__ == "__main__":
    main()